### PR TITLE
Update softmax_output.cc

### PR DESCRIPTION
**Pull request by kli-nlpr**
I think that 'Computes the gradient of cross entropy loss with respect to softmax output' is confusing.
So I change it to
'Computes the gradient of cross entropy loss with respect to the input array `data` '

https://math.stackexchange.com/questions/945871/derivative-of-softmax-loss-function

### CODE REVIEW

File: src/operator/softmax_output.cc [hunk 1]
```diff
@@ -62,7 +62,7 @@ MXNET_REGISTER_OP_PROPERTY(SoftmaxOutput, SoftmaxOutputProp)
 
     .. math:: \text{CE(label, output)} = - \sum_i \text{label}_i \log(\text{output}_i)
 
-  - The gradient of cross entropy loss w.r.t softmax output:
+  - The gradient of cross entropy loss w.r.t the input array:
 
     .. math:: \text{gradient} = \text{output} - \text{label}
```
Comment (piiswrong) [inline src/operator/softmax_output.cc]: This is indeed describing the loss w.r.t. the output, not input

Comment (chinakook) [reply src/operator/softmax_output.cc]: As I've searched, the gradient will give a correction(i.e. gradient) which can be represented by the softmax activation to the features before softmax output(activation), not the softmax output(activation).
https://math.stackexchange.com/questions/945871/derivative-of-softmax-loss-function

Comment (kli-nlpr) [reply src/operator/softmax_output.cc]: @piiswrong  As @chinakook  said, I think it is the gradient of the loss wrt the input of the softmax activation function

**Comment by piiswrong** [comment]
closing due to inactive
